core/commands/IssueRelationshipAddCommand.php

```diff
@@ -0,0 +1,206 @@
+		$t_target_issue_ref = $this->payload( 'issue' );
+
+		if( !isset( $t_target_issue_ref['id'] ) ) {
+			throw new ClientException(
+				'Invalid issue id',
+				ERROR_INVALID_FIELD_VALUE,
+				array( 'issue_id' )
+			);
+		}
+
+		$t_target_issue_id = helper_parse_issue_id( $t_target_issue_ref['id'], 'target_issue_id' );
+
+		$this->sourceIssue = bug_get( $t_source_issue_id, true );
+
+		$t_update_threshold = config_get( 'update_bug_threshold', null, null, $this->sourceIssue->project_id );
+
+		# Ensure user has access to update the source issue
+		if( !access_has_bug_level( $t_update_threshold, $t_source_issue_id ) ) {
+			throw new ClientException(
+				'Access denied to add relationship',
+				ERROR_ACCESS_DENIED
+			);
+		}
+
+		# Ensure that source and target issues are not the same
+		if( $t_source_issue_id == $t_target_issue_id ) {
+			throw new ClientException(
+				"Issue can't have relationship to itself",
+				ERROR_RELATIONSHIP_SAME_BUG
+			);
+		}
+
+		# Ensure that related issue exists and gets its information
+		$this->targetIssue = bug_get( $t_target_issue_id, true );
+
+		# Ensure source issue is not read-only
+		if( bug_is_readonly( $t_source_issue_id ) ) {
+			throw new ClientException(
+				sprintf( "Issue %d is read-only", $t_source_issue_id ),
+				ERROR_BUG_READ_ONLY_ACTION_DENIED,
+				array( $t_source_issue_id )
+			);
+		}
+
+		# Ensure that user can view target issue
+		$t_view_threshold = config_get( 'view_bug_threshold', null, null, $this->targetIssue->project_id );
+		if( !access_has_bug_level( $t_view_threshold, $t_target_issue_id ) ) {
+			throw new ClientException(
+				sprintf( "Issue %d is read-only", $t_target_issue_id ),
+				ERROR_RELATIONSHIP_ACCESS_LEVEL_TO_DEST_BUG_TOO_LOW,
+				array( $t_target_issue_id )
+			);
+		}
+	}
+
+	/**
+	 * Process the command.
+	 *
+	 * @returns array Command response
+	 */
+	protected function process() {
+		if( $this->sourceIssue->project_id != helper_get_current_project() ) {
+			# in case the current project is not the same project of the bug we are
+			# viewing, override the current project. This to avoid problems with
+			# categories and handlers lists etc.
+			global $g_project_override;
+			$g_project_override = $this->sourceIssue->project_id;
+		}
+
+		# Create or update the relationship
+		$t_relationship_id = relationship_upsert( $this->sourceIssue->id, $this->targetIssue->id, $this->typeId );
+
+		return array( 'id' => $t_relationship_id );
+	}
+
+	/**
+	 * Get relationship type id from relationship type reference.
+	 *
+	 * @param array The relationship type reference with id, name or both.
+	 * @return integer relationship type id.
+	 */
+	private function getRelationTypeId( $p_relationship_type ) {
+		if( isset( $p_relationship_type['id'] ) ) {
+			$t_type_id = (int)$p_relationship_type['id'];
+		} else if( isset( $p_relationship_type['name'] ) ) {
+			global $g_relationships;
+			switch( $p_relationship_type['name'] ) {
+				case 'related-to':
```

Comment:
In the SOAP API we had a pattern where an entity can be referenced by its id or name.  If both are supplied, the id is used.

In the REST API, we use the same model.  However, when getting back a reference from the server, we often including extra information like `label` which is the localized language based on the user that issued the request.  That applies across the board.

It is a core part of the REST API design to not just dump internal structures to users and expects client apps to construct them when issuing requests.

Reply:
I have done some refactoring and fixing to existing code to address the following:

- Have the mapping in the relationship APIs and make it consistent with other similar transformations.
- Fix an issue where issue get was localizing the name rather than having a label field. Also added label field there.